service/api/service.go

```diff
@@ -134,18 +152,37 @@ func (s *serviceImpl) ApiV1WorkflowStartPost(
 		InitSearchAttributes:               initCustomSAs,
 		InitDataAttributes:                 initCustomDAs,
 		Config:                             workflowConfig,
-		UseMemoForDataAttributes:           useMemo,
+		UseMemoForDataAttributes:           useMemoForDAs,
 		WaitForCompletionStateExecutionIds: req.GetWaitForCompletionStateExecutionIds(),
 		WaitForCompletionStateIds:          req.GetWaitForCompletionStateIds(),
 		OmitVersionMarker:                  s.config.Api.OptimizedVersioning,
 	}
 
 	runId, err := s.client.StartInterpreterWorkflow(ctx, workflowOptions, input)
 	if err != nil {
-		return nil, s.handleError(err, WorkflowStartApiPath, req.GetWorkflowId())
-	}
+		shouldHandleError := true
 
-	s.logger.Info("Started workflow", tag.WorkflowID(req.WorkflowId), tag.WorkflowRunID(runId))
+		if s.client.IsWorkflowAlreadyStartedError(err) && ignoreAlreadyStartedError {
+			response, descErr := s.client.DescribeWorkflowExecution(ctx, req.GetWorkflowId(), "", nil)
+			if descErr != nil {
+				return nil, s.handleError(err, WorkflowStartApiPath, req.WorkflowId)
+			}
+
+			runId = response.RunId
```

Comment:
This shouldn't be different. I think we should pass the runId that returned from `StartInterpreterWorkflow` into `DescribeWorkflowExecution`. This is to be more accurate and avoid some edge cases from happening.

Reply:
Sure. Didn't know that that error would return `runId`. Addressed